tools/wptrunner/wptrunner/executors/executorselenium.py

```diff
@@ -196,12 +198,31 @@ def do_test(self, test):
         return (test.result_cls(*data), [])
 
     def do_testharness(self, webdriver, url, timeout):
-        return webdriver.execute_async_script(
-            self.script % {"abs_url": url,
-                           "url": strip_server(url),
-                           "window_id": self.window_id,
-                           "timeout_multiplier": self.timeout_multiplier,
-                           "timeout": timeout * 1000})
+        format_map = {"abs_url": url,
+                      "url": strip_server(url),
+                      "window_id": self.window_id,
+                      "timeout_multiplier": self.timeout_multiplier,
+                      "timeout": timeout * 1000}
+        webdriver.execute_script(self.script % format_map)
+        while True:
+            result = webdriver.execute_async_script(
+                self.script_resume % format_map)
+            if result[1] == "complete":
+                break
+            elif result[1] == "action":
+                parent = webdriver.current_window_handle
+                try:
+                    webdriver.switch_to.window(self.window_id)
+                    action = result[2]["action"]
+                    if action == "click":
+                        selector = result[2]["selector"]
+                        elements = webdriver.find_elements_by_css_selector(selector)
+                        assert len(elements) == 1  # the JS should ensure this
+                        elements[0].click()
```

Comment:
I wrote in the original API spec in the first commit that we'd reject the promise, but that again requires some way of communicating back from the thing that triggered the JS callback, ergh.